### ISSUE TITLE
Phase D Step 6: Backwards-compat: use_build_order_reward flag default false

### ISSUE THREAD

**Issue by aberson** [aberson/Alpha4Gate#169]
## Build step 6 of 9 — Phase D (umbrella #114)

**Build doc:** [documentation/plans/phase-d-build-plan.md](../blob/master/documentation/plans/phase-d-build-plan.md) §3 Step D.6

### What to build
Add hyperparams `use_build_order_reward: false` and `build_order_reward_alpha: 1.0` to `hyperparams.json`. When the flag is false, reward computation is byte-identical to baseline. When true, `RewardCalculator` also calls `build_order_reward.step_reward(...)` (scaled by `alpha`) and adds the result.

- Add `"use_build_order_reward": false` and `"build_order_reward_alpha": 1.0` to `bots/current/data/hyperparams.json`
- Extend `RewardCalculator.__init__` to read both values from a passed-in hyperparams dict (or load directly if `rules_path`'s sibling `hyperparams.json` exists)
- Per-step path: if flag is true AND `GameSnapshot.current_build_order is not None`, compute `step_reward(prev_progress, curr_progress, alpha)` and add to the per-step total. `prev_progress` is initialized to `None` at game start; first step uses 0
- `RewardCalculator` owns the progress state across steps (resets in `open_game_log`)
- Per-step JSONL log lines gain an optional `build_order_reward` numeric field when the flag is on (`null` or absent when off, to keep flag-off byte-identical)

### Existing context
- `hyperparams.json` is a flat dict ([hyperparams.json:1-16](../blob/master/bots/v13/data/hyperparams.json#L1-L16)). Current keys (14): `learning_rate`, `n_steps`, `batch_size`, `n_epochs`, `gamma`, `gae_lambda`, `clip_range`, `ent_coef`, `vf_coef`, `max_grad_norm`, `net_arch`, `policy_type`, `kl_rules_coef`, `use_imitation_init`. No nesting; the two new keys live alongside existing entries.
- `RewardCalculator` is constructed once per game in the runner; per-step state ownership is fine.
- Per-step JSONL line shape today (written at [rewards.py:213-221](../blob/master/bots/v13/learning/rewards.py#L213-L221)):
  ```json
  {"game_time": 245.3, "total_reward": 0.187, "fired_rules": ["scout-early"], "is_terminal": false, "result": null}
  ```
  D.6 adds the optional `build_order_reward` field.

### Files to modify/create
- `bots/current/data/hyperparams.json` — add the two flags
- `bots/current/learning/rewards.py` — read flags, wire summand. Update the inline header comment to mention the new flags
- `tests/test_reward_flag.py` (NEW)

### Done when
- Flag-off path produces byte-identical per-step rewards to baseline on a recorded game log (use the same fixture as D.4's parity test)
- Flag-on path adds the build-order summand correctly on the same log (assert nonzero delta)
- `tests/test_rewards.py` still passes (no regression)
- `uv run pytest tests/test_reward_flag.py tests/test_rewards.py tests/test_reward_migration.py` all pass

### Flags (recommended)
`--reviewers code --isolation worktree`

### Depends on
#168 (Phase D Step 5) — needs the `current_build_order` field on `GameSnapshot` to gate the flag-on path.

### Parallel-safe with
None within Phase D — D.6 depends on D.5; D.6.5 (#270) depends on D.6; M1 (#170) depends on D.6.

### Produces
Hyperparams + rewards.py extension + test file.

### Operator workflow notes (autonomous + UI-bundle + parallel)
- Backend-only flag wiring; no `--ui` flag.
- Default off — byte-identical baseline preserved. M1 (#170) flips the flag for the training cycle.

---
*Synced from [documentation/plans/phase-d-build-plan.md](../blob/master/documentation/plans/phase-d-build-plan.md) by /repo-sync at d2a19da (plan edits uncommitted; commit after sync)*

**Comment by aberson** [comment]
## build-phase: Step D.6 started
**Step:** Backwards-compat use_build_order_reward flag
**Phase:** D
**Flags:** --reviewers code --isolation worktree
**Baseline tests:** 1657 after D.1-D.5 merged at c97f522
**Depends on:** D.5 PASS (#168 closed)
**Started:** 2026-05-20

**Comment by aberson** [comment]
## build-phase: Step D.6 complete

**Verdict:** PASS (iteration 1/3)

### Changes made
- **bots/v13/data/hyperparams.json** — added `"use_build_order_reward": false` and `"build_order_reward_alpha": 1.0` (16 flat keys total, defaults OFF).
- **bots/v13/learning/rewards.py** — extended RewardCalculator:
  - `__init__(hyperparams=None)` kwarg with sibling-file fallback load.
  - `_reset_build_order_state()` clears `_prev_progress`, `_active_build_order`, `_executed_actions`, `_prev_counts`. Called from `__init__` AND `open_game_log`.
  - `_build_order_summand()` — lazy-loads trajectory via `load_build_order(snapshot.current_build_order)`, accumulates executed_actions across steps, computes `compute_progress` + `step_reward(prev, curr, alpha)`.
  - `_COUNT_FIELD_TO_ACTION` (21 entries) maps GameSnapshot count fields to (action, target) tuples. When a count goes up by N, N events appended. Covers: gateway, robo→roboticsfacility, forge, cannon, battery, base→nexus, + 15 unit-train counts.
  - Per-step JSONL log writes `build_order_reward: <float>` ONLY when flag is ON AND summand was computed (field absent when off — byte-identical to baseline).
- **bots/v13/learning/hyperparams.py** — added the two new keys to `_TRAINING_KWARGS` so `to_ppo_kwargs` filters them out before passing to PPO() (silent-wiring fix per memory feedback_buildstep_require_integration_test; NOT in plan's Files list — informational).
- **tests/test_reward_flag.py** (NEW, 11 tests) — cardinal invariant (flag-off byte-identical), flag-on summand added, alpha scaling, state reset, JSONL field presence/absence round-trip.

### Cardinal invariant: VERIFIED
When `use_build_order_reward=False`:
- `_build_order_summand` returns None early.
- `compute_step_reward` adds nothing extra.
- JSONL log line is byte-identical to baseline (no `build_order_reward: null`, no `: 0.0` — field absent).
Pinned by 4 tests including a 200-step synthetic sequence comparison.

### Known signal-quality limitation (MEDIUM, surfaced for M1)
The dev derived `executed_actions` from positive deltas in GameSnapshot count fields. GameSnapshot has count fields for gateway, roboticsfacility, forge, cannon, battery, nexus, and 15 unit types — but NOT for pylon, assimilator, cyberneticscore, roboticsbay, warp_gate_research. These 5 trajectory targets will permanently appear as "missed" in `compute_progress`, applying a constant negative offset whenever the flag is on.

Impact: ~5/11 4-gate-aggression targets unreachable, ~5/12 robo-colossus targets unreachable. The flag-on signal is meaningfully diluted but NOT zero — gateway, roboticsfacility, observer, forge, colossus, zealot, stalker, nexus do produce real differential signal.

Mitigation: flag is OFF in production (verified by `test_default_flag_is_off`). M1's variance-reduction measurement is positioned to discover whether the signal is usable as-is. If M1 reveals the dilution is too severe, follow-up issue: add pylon_count / assimilator_count / cyberneticscore_count / roboticsbay_count to GameSnapshot.

### Pre-existing test failure (NOT a D.6 regression)
`tests/test_current_pointer.py::test_dash_m_submodule_runs_via_alias` — `python -m bots.current.runner --help` subprocess gets sys.argv[0] = None, argparse trips. Confirmed identical failure on stashed (pre-D.6) state.

### Reviewers
- Correctness: APPROVED (1 MEDIUM on signal dilution acknowledged + 1 LOW cosmetic on redundant exception catch).
- Bug: APPROVED (cardinal invariant verified, state reset clean, hyperparams.py drop-from-PPO verified, JSONL shape pinned, side effects clean).
- Test Quality: APPROVED (all 11 tests load-bearing, no duplicates, JSONL round-trip covered).

**Files changed:** 4 (3 modified: hyperparams.json, rewards.py, hyperparams.py; 1 new test)
**Tests:** 1657 → 1668 (+11)
**Gates:** typecheck PASS (802 files), lint PASS, test PASS

**Duration:** ~22m
